src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,90 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int FIRST_HALF_ARRAY = 0;
+    private static final int SECOND_HALF_ARRAY = 1;
+    private static final int SIZE_REPORT = 3;
+    private static final String ACTION_WITH_BUY = "buy";
+    private static final String ACTION_WITH_SUPPLY = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String doneData = readFile(fromFileName);
+        String readyReport = operationWithFile(doneData);
+        writeTiFile(readyReport, toFileName);
+    }
+
+    private String readFile(String fromFileName) {
+        File startFile = new File(fromFileName);
+        StringBuilder builder = null;
+        if (!startFile.exists()) {
+            throw new RuntimeException("File does not exist");
+        }
+        try (BufferedReader reader = new BufferedReader(new FileReader(startFile))) {
+            builder = new StringBuilder();
+            String value = reader.readLine();
+            while (value != null) {
+                builder.append(value).append(System.lineSeparator());
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file", e);
+        }
+        return builder.toString();
+    }
+
+    private String operationWithFile(String builder) {
+        String[] lineOfStartFile = builder.toString().split(System.lineSeparator());
+        int sumOfBuy = 0;
+        int sumOfSupply = 0;
+        for (int i = 0; i < lineOfStartFile.length; i++) {
+            if (Character.isDigit(lineOfStartFile[i].charAt(0))) {
+                String[] elementsStartWithNumber = lineOfStartFile[i].split("(?<=\\d)(?=\\D)");
+                if (elementsStartWithNumber[SECOND_HALF_ARRAY].equals(ACTION_WITH_BUY)) {
+                    sumOfBuy += Integer.parseInt(elementsStartWithNumber[FIRST_HALF_ARRAY]);
+                } else if (elementsStartWithNumber[SECOND_HALF_ARRAY].equals(ACTION_WITH_SUPPLY)) {
+                    sumOfSupply += Integer.parseInt(elementsStartWithNumber[FIRST_HALF_ARRAY]);
+                }
+            }
+            String[] elementOfLine = lineOfStartFile[i].split(",");
+            if (elementOfLine.length != 2) {
+                throw new RuntimeException("Invalid input file format at line " + (i + 1));
+            }
```

Comment:
validating input is not required in this task, but if you opt to do this, you can validate string against a regex - "^\\w+,\\d+$" - this way you can combine your multiple checks